Apps/WebClient/src/NewClientApp/src/components/private/accept-terms-of-service/AcceptTermsOfServiceView.vue

```diff
@@ -0,0 +1,132 @@
+
+const isLoading = ref(true);
+const termsOfServiceId = ref("");
+const termsOfService = ref("");
+const accepted = ref(false);
+
+function onSubmit(): void {
+    if (accepted.value !== true) {
+        return;
+    }
+
+    isLoading.value = true;
+
+    userStore
+        .updateAcceptedTerms(termsOfServiceId.value)
+        .then(() => redirect())
+        .catch((err: ResultError) => {
+            logger.error(
+                `Error updating accepted terms of service: ${JSON.stringify(
+                    err
+                )}`
+            );
+            if (err.statusCode === 429) {
+                errorStore.setTooManyRequestsError("page");
+            } else {
+                errorStore.addError(
+                    ErrorType.Update,
+                    ErrorSourceType.Profile,
+                    err.traceId
+                );
+            }
+        })
+        .finally(() => {
+            isLoading.value = false;
+        });
+}
+
+function redirect(): void {
+    logger.debug("Redirecting to /home");
+    router.push({ path: "/home" });
+}
+
+// Created hook
+userProfileService
+    .getTermsOfService()
+    .then((result) => {
+        logger.debug(`getTermsOfService result: ${JSON.stringify(result)}`);
+        termsOfServiceId.value = result.id;
+        termsOfService.value = result.content;
+    })
+    .catch((err: ResultError) => {
+        logger.error(err.resultMessage);
+        if (err.statusCode === 429) {
+            errorStore.setTooManyRequestsWarning("page");
+        } else {
+            errorStore.addError(
+                ErrorType.Retrieve,
+                ErrorSourceType.TermsOfService,
+                undefined
+            );
+        }
+    })
+    .finally(() => {
+        isLoading.value = false;
+    });
+</script>
+
+<template>
+    <LoadingComponent :is-loading="isLoading" />
+    <PageTitleComponent
+        title="Update to our Terms of Service"
+        data-testid="tos-page-title"
+    />
+    <template v-if="!isLoading && termsOfService !== ''">
+        <HtmlTextAreaComponent
+            class="termsOfService mb-4"
+            :input="termsOfService"
+            data-testid="tos-text-area-component"
+        />
+        <div class="mb-4">
+            <v-checkbox
+                id="accept-tos-checkbox"
+                v-model="accepted"
+                color="primary"
+                data-testid="accept-tos-checkbox"
+                label="I agree to the terms of service above"
+                :error-messages="
+                    accepted ? [] : ['You must accept the terms of service.']
+                "
+            />
+        </div>
+        <div class="text-right">
+            <HgButtonComponent
+                data-testid="continue-btn"
+                variant="primary"
+                :disabled="!accepted"
+                text="Continue"
+                @click.prevent.stop="onSubmit"
+            ></HgButtonComponent>
```

Comment:
```suggestion
                @click="onSubmit"
            />
```
I don't think the modifiers are needed, since there's no longer a form element.